spec/rspec/core/formatters/snippet_extractor_spec.rb

```diff
@@ -250,6 +250,24 @@ def another_expression(*)
         end
       end
 
+      context 'when Ripper cannot parse the source', :isolated_directory do
+        let(:file_path) do
+          'invalid_source.rb'
+        end
+
+        let(:line_number) do
+          1
+        end
+
+        before do
+          File.write(file_path, 'invalid_source]')
+        end
+
+        it 'returns the line by falling back to the simple single line extraction' do
+          expect(expression_lines).to eq(['invalid_source]'])
+        end
+      end
```

Comment:
This spec tests a situation that, as far as I can tell, can't happen in real life: a spec file with invalid syntax could never get run by RSpec and therefore could never be the source of a failure.  I understand the need to write this kind of test for the JRuby thing but nothing in the test mentions the JRuby bug, so future me will be confused about why this behavior is necessary.  WDYT about adding an explanatory note to the `context`?

``` ruby
context 'when Ripper cannot parse the source (which can happen on JRuby -- see jruby/jruby#2427)', :isolated_directory do
  # ...
end
```

This makes the need for the test much more explicit.